sstables/sstable_set_impl.hh

```diff
@@ -60,6 +61,7 @@ private:
     interval_map_type _leveled_sstables;
     lw_shared_ptr<sstable_list> _all;
     std::unordered_map<utils::UUID, sstable_run> _all_runs;
+    std::unordered_map<generation_type, shared_sstable> _mapped;
```

Comment:
do we need it to be a map? cannot it be std::unordered_set<generation_type> instead?

Reply:
Hmm, I think it can be.

Reply:
That's the direction of https://github.com/bhalevy/scylla/tree/unique_genration_sstable_set
`using unique_genration_sstable_set = std::unordered_set<shared_sstable, sstable_gen_hash, sstable_gen_equal_to>`